src/compiler/parser.ts

```diff
@@ -6728,6 +6794,26 @@ namespace ts {
                     return currentToken = scanner.scanJSDocToken();
                 }
 
+                function parseJSDocEntityName(createIfMissing = false): EntityName {

```

Comment:
Done. I thought about merging `parseEntityName` and `parseJSDocEntityName`, but they both worry about ad-hoc jsdoc-related things, plus they call into different identifier parsers which do the same.